proxy/grpcproxy/cache/store.go

```diff
@@ -25,8 +25,12 @@ import (
 )
 
 var (
-	DefaultMaxEntries = 2048
-	ErrCompacted      = rpctypes.ErrGRPCCompacted
+	DefaultMaxEntries   = 2048
+	ErrCompacted        = rpctypes.ErrGRPCCompacted
+	MaxRangeRequestSize = 1024 * 4
+	// a temporary space used for Marshalling the RangeRequest for cache keyFunc
+	// should be used in keyFunc function only
+	sliceRangeRequestMarshal = make([]byte, MaxRangeRequestSize)
```

Comment:
this should be in `cache`, otherwise there will be a data race if there are two concurrent `cache`s